jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/JakartaLanguageServer.java

```diff
@@ -70,8 +71,10 @@ public synchronized void updateSettings(Object initializationOptionsSettings) {
 
     @Override
     public CompletableFuture<Object> shutdown() {
-        // TODO Auto-generated method stub
-        return null;
+        // when shutting down LS, TextDocumentService.didClose() may not be called
+        // properly, need to clear existing diagnostics
+        ((JakartaTextDocumentService) textDocumentService).cleanDiagnostics();
+        return CompletableFutures.computeAsync(cc -> new Object());
```

Comment:
Usually we use `()` when there are no arguments. A variable name may imply something is missing.

Reply:
not sure what do you mean

Reply:
`cc -> new Object()` defines an anonymous class with a method that takes the argument `cc` but that argument is not used in the body of the method `new Object()`. This can be confusing when trying to maintain the code especially when the method is longer, non-trivial. Instead, the usual way to code a lambda that does not take any arguments is `() -> new Object()'. This way people do not look for where the argument is used and they don't need to wonder if the part of the method that uses an argument like `cc` has been accidentally erased.

Reply:
I don't know how much confusing this one line code can cause here, if you are talking about changing something like `() -> new Object()`, unfortunately, in Java, you can NOT ignore pre-defined argument:
![image](https://user-images.githubusercontent.com/58234271/199316975-27281e99-2ec8-4ce1-a7b8-29babd43dc94.png)

Reply:
Ok, sorry, I didn't see that message when I tried it.

Reply:
Heh, just realized CompletableFutures is not CompletableFuture. Tricky. Rookie mistake.